app/views/users/show.html.erb

```diff
@@ -8,4 +8,13 @@
       </h1>
     </section>
   </aside>
+  <div class="col-md-8">
+    <% if @user.microposts.any? %>
+      <h3>Microposts (<%= @user.microposts.count %>)</h3>
```

Comment:
í8n, so sánh size, count, length

Reply:
Các method này được sử dụng cho array hoặc object.
trong aray:
+ length trong array sẽ trả về số phần tử trong mảng, size cũng tương tự length
+ count có thể sử dụng đi kèm điều kiện để tính toán số phần tử. có 3 cách gọi sau : 

> Array#count trả về số phần tử trong mảng

> Array#count n trả về số phần tử có giá trị = n trong mảng

> Array#count{|i| i.even?} trả về số phần tử trong mảng dựa trên điều kiện.

trong ActiveRecord:
 + cout: sẽ sd câu lệnh truy vấn sql để lấy ra số lượng bản ghi mà chúng ta truy vấn. cout không lưu trong object chúng ta gọi. mỗi lanaf gọi đến cout thì nó sec thực hiện câu lênh truy vấn vào database. dù trước đó chúng ta có gọi nó hay chưa.
+ length: nó sẽ đếm số lượng mà không sử dụng thêm câu truy vấn nếu đối tượng đó đã được nạp, ngược lại, nó sẽ nạp toàn bộ object vào bộ nhớ và thực hiện đếm. nên khi gọi length từ lần thứ 2trowr đi nó sec không truy vấ vào databasse nữa mà sẽ trả về luôn kết quả.
+ size: nó sec truy vấn vào database khi đối tượng chưa được nạp, còn đối tượng đã được nạp thì nó sec đếm luôn trong bộ nhớ chứ không truy vấn vào data base nữa
+